crates/iceberg-catalog/src/implementations/postgres/user.rs

```diff
@@ -0,0 +1,270 @@
+use super::dbutils::DBErrorHandler;
+use crate::api::management::v1::{
+    CreateRoleRequest, ListRolesResponse, ListUsersResponse, Role, UpdateUserRequest, User,
+    UserOrigin,
+};
+use crate::service::token_verification::UserId;
+use crate::service::{ErrorModel, Result};
+use sqlx::PgConnection;
+
+#[derive(sqlx::Type, Debug, Clone, Copy)]
+#[sqlx(rename_all = "kebab-case")]
+enum DbUserOrigin {
+    ExplicitViaRegisterCall,
+    ImplicitViaConfigCall,
+}
+#[derive(sqlx::FromRow, Debug)]
+pub struct UserRow {
+    id: String,
+    name: String,
+    email: Option<String>,
+    origin: DbUserOrigin,
+    created_at: chrono::DateTime<chrono::Utc>,
+    updated_at: Option<chrono::DateTime<chrono::Utc>>,
+}
+
+impl From<UserRow> for User {
+    fn from(
+        UserRow {
+            id,
+            name,
+            email,
+            origin,
+            created_at,
+            updated_at,
+        }: UserRow,
+    ) -> Self {
+        User {
+            id,
+            name,
+            email,
+            user_origin: match origin {
+                DbUserOrigin::ExplicitViaRegisterCall => UserOrigin::ExplicitViaRegisterCall,
+                DbUserOrigin::ImplicitViaConfigCall => UserOrigin::ImplicitViaConfigCall,
+            },
+            created_at,
+            updated_at,
+        }
+    }
+}
+
+pub(crate) async fn list_users<'e, 'c: 'e, E: sqlx::Executor<'c, Database = sqlx::Postgres>>(
+    connection: E,
+    include_deleted: bool,
+    name_filter: Option<&str>,
+) -> Result<ListUsersResponse> {
+    // TODO: impl pagination
+    let users = sqlx::query_as!(
+        UserRow,
+        r#"
+        SELECT
+            id,
+            name,
+            origin as "origin: DbUserOrigin",
+            email,
+            created_at,
+            updated_at
+        FROM users
+        where (deleted_at is null OR $1) AND (name ILIKE $2)
```

Comment:
I don't think using ILIKe on user provided input is a good idea.
We should at least limit search string length and make sure it doesn't contain (or escape) "%" (and other reserved chars?).
We should also add %<search-term>% or something like this by default.
If I am looking for a part of a last name I would still expect to find the person even if I don't specify a full regex.

Maybe there is a certain index in Postgres for free text search?